### PR TITLE
Add MacOS and Windows build targets

### LINKED ISSUE #555: MacOS and windows pre-compiles binaries
pydantic will soon be compiled with cython, however pre-compiled binaries are only available on linux at the moment.

It should be possible for macos on travis and windows on azure pipeline.

However since I only use linux, it's not my primary concern.

I'd be happy to accept a PR if someone is prepared to implement one or both.

### PR DESCRIPTION
**Pull request by StephenBrown2**
## Change Summary

As of about a week ago, the cibuildwheel repository has new examples for Travis on Mac and Windows: https://github.com/joerick/cibuildwheel/blob/master/examples/travis-ci-test-and-deploy.yml

This adds those targets to .travis.yml to hopefully resolve #555 .

If this passes (which I can't test myself, since you should handle the releases), I can edit https://github.com/samuelcolvin/pydantic/blob/8727914ced9d4546f0f95a453aff78f03b778275/docs/install.md#L19-L21 to reflect it.

## Related issue number

#555 

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Humm, I'd rather move all this to github actions, see [rtoml](https://github.com/samuelcolvin/rtoml) for a more complex but similar case.

Still until we move to github actions completely, maybe this would be a good stop gap.

**Comment by samuelcolvin** [comment]
please confirm that these steps are working by commenting out the `if: type = push AND (branch = master OR tag IS present)` and `deploy:` chunks.

**Comment by StephenBrown2** [comment]
Well, MacOS is building, but I can't figure out the Windows errors at the moment.  I will happily leave those commented out for someone else to tackle, if you wish.

**Comment by demospace** [comment]
> Well, MacOS is building, but I can't figure out the Windows errors at the moment. I will happily leave those commented out for someone else to tackle, if you wish.

I patched pydantic in a few places recently and tried to compile in Windows and ran into the following:

`\temp.win-amd64-3.7\Release\pydantic\__init__.cp37-win_amd64.lib
LINK : error LNK2001: unresolved external symbol PyInit___init__
build\temp.win-amd64-3.7\Release\pydantic\__init__.cp37-win_amd64.lib : fatal error LNK1120: 1 unresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120
`
The issue relates to distutils/setuptools and you can read more about it here:
 https://bugs.python.org/issue35893

The last comment in that thread, when added to pydantic's setup.py does resolve the issue:
`from distutils.command import build_ext

def get_export_symbols(self, ext):
    parts = ext.name.split(".")
    print('parts', parts)
    if parts[-1] == "__init__":
        initfunc_name = "PyInit_" + parts[-2]
    else:
        initfunc_name = "PyInit_" + parts[-1]`

**Comment by StephenBrown2** [comment]
In theory this could break non-windows? So best to include it in an `os.name` check, like:
```py
import os
from distutils.command import build_ext

if os.name == 'nt':

    def pass_ges(self, ext):
        pass

    build_ext.get_export_symbols = pass_ges
```

**Comment by samuelcolvin** [comment]
> In theory this could break non-windows? So best to include it in an `os.name` check, like:
> 
> ```python
> import os
> from distutils.command import build_ext
> 
> if os.name == 'nt':
> 
>     def pass_ges(self, ext):
>         pass
> 
>     build_ext.get_export_symbols = pass_ges
> ```

agreed, let's try this.

**Comment by StephenBrown2** [comment]
Doesn't looke like it's fixed that issue:
```
  LINK : error LNK2001: unresolved external symbol PyInit___init__
```
still happening.

File: setup.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 from pathlib import Path
 
 from setuptools import setup
 
+if os.name == 'nt':
+
+    def pass_ges(self, ext):
+        pass
+
+    build_ext.get_export_symbols = pass_ges
```
Comment (samuelcolvin) [inline setup.py]: according to https://bugs.python.org/issue35893 this should be 

```py
build_ext.build_ext.get_export_symbols = ...
```

Also they have a different function for `get_export_symbols`.

Comment (StephenBrown2) [reply setup.py]: Good point, I didn't look at the bug. The Stackoverflow answer is what suggested the pass (as the "simplest" option), but I'll try the solution in the bug.

**Comment by StephenBrown2** [comment]
Well it looks like that indeed worked. I took the function from cpython master and the patch from the bug report, and Windows builds are now succeeding!

**Comment by samuelcolvin** [comment]
incorporated into #1326, thank you so much.

**Pull request closed without merging**